pkg/apis/core/validation/cloudprofile.go

```diff
@@ -100,6 +158,16 @@ func validateKubernetesSettings(kubernetes core.KubernetesSettings, fldPath *fie
 		} else {
 			versionsFound.Insert(version.Version)
 		}
+		allErrs = append(allErrs, validateExpirableVersion(version, idxPath)...)
+	}
+
+	return allErrs
+}
+
+func validateExpirableVersion(version core.ExpirableVersion, fldPath *field.Path) field.ErrorList {
```

Comment:
I think that is up to the operator to decide. If using the VersionManagement or the VersionMaintenance that is already made sure through reconciliation.

Reply:
Hm, but what if we want to phase out a minor versions, 1.10 for example? We would only specify deprecated versions, no supported ones. Would that work?

Reply:
When using no VersionManagement or VersionManagement, the operator can specify the versions without restrictions.

Using VersionManagement, the reconcile enforces one supported version independent of the minor version.

Using also VersionMaintenance, the reconcile enforces one supported version per maintained minor version. That means you cannot phase out maintained minor versions. However the number of maintained minor versions can be adjusted on the fly in order to make any minor version unmaintained. 
Does that make sense?

Reply:
> Using VersionManagement, the reconcile enforces one supported version independent of the minor version.

How do you phase out a version with this restriction?

Reply:
You can phase out any version using VersionManagement? As long as it is not the latest non-preview patch version of the latest minor version.

Reply:
I don't get it. If we offer 1.10 and 1.11 and we don't want to offer 1.10 anymore, how do we do it? You are saying that you enforce at least one supported version. However, what we want is to set an expiration date to every patch version of 1.10 so that we can delete it from the `CloudProfile` after all expiration dates are expired.

Reply:
Given you example, the  CloudProfile only contains minor versions 1.10 and 1.11. We want to expire all version from minor 1.10. 

Scenario 
1) using VersionManagement.
All but the latest patch version from 1.10 are deprecated automatically.  The operator needs to explicitly set an expiration date to the latest patch version as a safety measure.

2) VersionMaintenance. 
- Need to adjust the configuration of the CloudProfile controller so that 1.10 is an unmaintained minor version. 
- then it is the same as VersionManagement.

Is that more clear now?

Reply:
Not really. Again, you said "Using VersionManagement, the reconcile enforces one supported version independent of the minor version." (see https://github.com/gardener/gardener/pull/2105#discussion_r403992734). It contradicts to 100% the first point.

Reply:
OK, after a call with @danielfoehrKn it became clear that the statement above does **not** mean that there has to be one supported versions **per** minor version. Instead, there must be one supported version in all the versions.